WireAPI/Tests/WireAPITests/Helpers/RequestSnapshotHelper.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SnapshotTesting
+@testable import WireAPI
+import XCTest
+
+/// A helper object to make snapshotting API requests easier.
+
+struct RequestSnapshotHelper {
+    
+    /// Snapshot test the request generated by the given block for
+    /// all api versions.
+    /// 
+    /// This will generate one snapshot reference for each version.
+    ///
+    /// - Parameters:
+    ///   - block: Some code that should invoke a method of the given api to generate a request.
+    ///   - file: The file invoking the test.
+    ///   - function: The method invoking the test.
+    ///   - line: The line invoking the test.
+
+    func verifyRequestForAllAPIVersions(
+        when block: (any TeamsAPI) async throws -> Void,
+        file: StaticString = #file,
+        function: String = #function,
+        line: UInt = #line
+    ) async throws {
+        for apiVersion in APIVersion.allCases {
+            try await verifyRequest(
+                apiVersion: apiVersion,
+                when: block,
+                file: file,
+                function: function,
+                line: line
+            )
+        }
+    }
+
+    /// Snapshot test the request generated by the given block for a
+    /// specific api version.
+    ///
+    /// Note: this is run on the main thread as a workaround of a crash in
+    /// the SnapshotTesting framework when snapshotting in async environments.
+    /// See https://github.com/pointfreeco/swift-snapshot-testing/issues/822
```

Comment:
The issue is open since the beginning of January.. 😬 will it be fixed? with the `@MainActor` attribute we avoid possibility of concurrency here.

While I see the usefulness here in this case I'm always a bit careful to embed third-party libraries.. what do others think?

Reply:
It's always a trade off and I think it comes down to a decision if it's worth it. We already use this snapshot library and it does feel helpful. Could we live without it? Probably. We could also try to fix the issue for them.

Reply:
True, we already have it in the project for view snapshots. I just wanted to raise the attention here, but for me it's acceptable too, thanks.